src/renderer/api/kube-object-menu-registry.ts

```diff
@@ -0,0 +1,33 @@
+import { observable } from "mobx"
+import React from "react"
+
+export interface KubeObjectMenuComponents {
+  MenuItem: React.ComponentType<any>;
+}
+
+export interface KubeObjectMenuRegistration {
+  kind: string;
+  apiVersions: string[];
```

Comment:
This allows extensions to have different components for different apiVersions (for example `v1alpha1` vs `v1`). It also allows extension to register multiple versions for a component.